### ISSUE TITLE
AboutDialog dynamic state is broken

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#589]
From https://github.com/phetsims/phet-io/issues/1590. Using AboutDialog helped us find a constraint that we didn't want in the state engine (that all dynamic element "roots" must be in the state), which that issue fixed. From here we should decide how to go with AboutDialog.

1. Why is AboutDialog in the state? Can't we exclude all sub elements from the state and be done? Why do we need to support this?
2. If this doesn't work, then we need to do some work to support this. Over in https://github.com/phetsims/phet-io/issues/1590 we made it so AboutDialog itself doesn't need to be in the state, and as a result the dialog is being cleared and created by the state engine correctly, but the show/hide piece is not working. You can see that in the state wrapper for any phet-io sim (like charges and fields).

When you apply this patch, you get the `show` method working in the downstream sim, but not the `hide` method. 

<details>

```diff
Index: js/PhetMenu.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/PhetMenu.js	(revision 2a6d34f1fb6e0b2083cbdfca8a3b6a77189e4aa9)
+++ js/PhetMenu.js	(date 1575680522970)
@@ -121,7 +121,13 @@
     Node.call( this );
 
     const aboutDialogCapsule = new PhetioCapsule( tandem => {
-      return new AboutDialog( sim.name, sim.version, sim.credits, Brand, sim.locale, phetButton, tandem );
+      const aboutDialog = new AboutDialog( sim.name, sim.version, sim.credits, Brand, sim.locale, phetButton, tandem );
+
+      try {
+        aboutDialog.show();
+      }
+      catch( e ) {}
+      return aboutDialog;
     }, [], {
       tandem: tandem.createTandem( 'aboutDialogCapsule' ),
       phetioType: PhetioCapsuleIO( DialogIO )

```

<details>

@samreid @chrisklus is there a paper trail on why the AboutDialog is in the state right now? Should we set all sub elements phetioState:false?

**Comment by zepumph** [comment]
@samreid, @chrisklus, @jonathanolson, and @zepumph investigated this today. The first challenge we ran into was the templating var assertion discussed at the end of https://github.com/phetsims/phet-io/issues/1586#issuecomment-561966658. Commits fixing that are above.

**Comment by zepumph** [comment]
We looked into solving the show/hide piece in general, for all dialogs. We came up with the conclusion that we didn't want to "box ourselves into a corner" by saying that we never want to allow dialogs to be shown via state. So we added support for show/hide state by moving show/hide logic to be a listener to the `isShowingProperty`. This way, if the isShowingProperty is stateful and set to `true`, then the dialog will be shown when the state is set on the downstream sim.

In general we still want Dialogs to be `phetioState: false` (especially since they are all modal). This is controlled with the `phetioState` option in Dialog. But still we are happy that if we want to support stateful dialogs we can. 

After the above commit, the PhET-iO team is happy with the level of instrumentation for Dialogs (and the AboutDialog). It would be good to get @pixelzoom to review this change before closing. Are there any issues that you see with how we modified Dialog's interals?

**Comment by zepumph** [comment]
The above commit (https://github.com/phetsims/phet-io-test-sim/commit/9f87a1c7e3b6e5dc6f81425b288cb51506dc4af8) is an exploration in making sure that PhetioCapsules support state in arbitratily nested PhET-iO elements. This mimics the behavior of AboutDialog, which itself is not stateful (`phetioState:false`), but has children with textProperties that are stateful.

**Comment by pixelzoom** [comment]
Sorry, I'm not qualified to review this issue. I have no experience with PhetioCapsule. And I don't understand why it's undesirable for AboutDialog to be stateful, and why its children must be stateful.

**Comment by zepumph** [comment]
>Sorry, I'm not qualified to review this issue. I have no experience with PhetioCapsule. And I don't understand why it's undesirable for AboutDialog to be stateful, and why its children must be stateful.

Sorry I wasn't clear. I am primarily interested in reviewing the changes to Dialog.js, which we did for PhET-iO ,but effect all Dialogs.js. Primarily https://github.com/phetsims/sun/commit/bd128eed4363d026b3d666f00fcda49db6bc601f where we moved show/hide logic from prototype methods to a listener on `isShowingProperty`. This helps us support state and interoperability for PhET-iO, but I'm also interested in some PhET brand eyes on it. I'm also happy to talk if that is still not clear.
